### PR TITLE
chore: update dependencies, license headers

### PR DESCRIPTION
**Pull request by kapilt**
update c7n-left to 0.2.0

update all dependencies.

### CODE REVIEW

File: poetry.lock [hunk 1]
```diff
@@ -125,79 +120,54 @@ crt = ["awscrt (==0.16.9)"]
...
+importlib-metadata = "^5.1"
+jsonschema = ">=4.18"
+python-dateutil = "^2.8.2"
+pyyaml = ">=5.4.0"
+tabulate = "^0.9.0"
+urllib3 = "<2.0"
+
+[package.source]
+type = "directory"
+url = "../../.pyenv/versions/3.11.3/envs/sinistral-dev/src/cloud-custodian"
```
Comment (kapilt) [inline poetry.lock:143]: this behavior is problematic

**Review by thisisshi — APPROVED** [review]

File: stacklet/client/sinistral/cli.py [hunk 2]
```diff
@@ -168,4 +168,9 @@ def login(ctx, username, password, *args, **kwargs):
 
 
 if __name__ == "__main__":
-    cli()
+    try:
+        cli()
+    except Exception:
+        import pdb, sys, traceback
+        traceback.print_exc()
+        pdb.post_mortem(sys.exc_info()[-1])
```
Comment (johnsca) [inline stacklet/client/sinistral/cli.py:176]: This seems helpful for debugging, but do we want it as the default behavior for end-users? Maybe a `--debug-errors` flag?

File: poetry.lock (hunk 1 above)
Comment (johnsca) [reply poetry.lock]: 😬 Yes it is.

**Comment by johnsca** [comment]
Output from the CI / Lint job is oddly unhelpful, but it looks like black is complaining about wanting an extra blank line, flake8 is complaining about the multi-import in the exception handler, and debug is complaing that pdb is imported at all.

File: stacklet/client/sinistral/cli.py (hunk 2 above)
Comment (johnsca) [reply stacklet/client/sinistral/cli.py:176]: Output from the CI / Lint job is oddly unhelpful, but it looks like it's all from this block.

black is complaining about wanting an extra blank line after the import, flake8 is complaining about the multi-import, and debug is complaing that pdb is imported at all.

File: poetry.lock [hunk 3]
```diff
@@ -284,24 +274,29 @@ pycparser = "*"
...
+version = "3.0.1"
+=======
+version = "3.2.0"
+>>>>>>> cc923ba (use the new c7n-left release)
 description = "The Real First Universal Charset Detector. Open, modern and actively maintained alternative to Chardet."
-category = "main"
 optional = false
 python-versions = ">=3.7.0"
 files = [
+<<<<<<< HEAD
```
Comment (johnsca) [inline poetry.lock:299]: Merge fragments breaking this.

File: stacklet/client/sinistral/client.py [hunk 4]
```diff
@@ -196,7 +196,10 @@ def make_request(
                 validate(_json, schema)
             res = func(path, q_params, _json)
             status_code = res.status_code
-            res = res.json()
+            try:
+                res = res.json()
+            except json.JSONDecodeError:
+                res = {"detail": print(res.text)}
```
[the remark this answers is not in the document]
Comment (kapilt) [reply stacklet/client/sinistral/client.py:202]: doh! thanks

File: stacklet/client/sinistral/cli.py (hunk 2 above)
Comment (kapilt) [reply stacklet/client/sinistral/cli.py:176]: end users go through cli entry points, this one only triggers if doing python -m stacklet.client.sinistral.cli

File: poetry.lock (hunk 3 above)
Comment (kapilt) [reply poetry.lock:299]: ugh.. thanks thought I had been able to wipe those by rm the lock, and doing an poetry install

**Pull request merged**